backend/controller/dal/testdata/go/fsmnext/fsmnext.go

```diff
@@ -0,0 +1,121 @@
+package fsmnext
+
+import (
+	"context"
+	"errors"
+
+	"github.com/TBD54566975/ftl/go-runtime/ftl"
+)
+
+func fsm() *ftl.FSMHandle {
+	// This FSM allows transitions moving forward through the alphabet
+	// Each transition also declares the next state(s) to transition to using State
+	//
+	//ftl:retry 2 1s
+	var fsm = ftl.FSM("fsm",
```

Comment:
Would it be better to introduce `ftl.FSMNext()` in this PR so we don't establish this pattern at all?

Reply:
@alecthomas I've added that now. Do you know why the grpc for a call included metadata key value pairs? I can't see it used anywhere, but this seemed like a reasonable use case for it so I piggy backed on it. What do you think?
Isolated commit here: [move from fsm.Next() to FSMNext()](https://github.com/TBD54566975/ftl/pull/1723/commits/20791c0d69401e8365a413e4f1d7cf380f462723)